modules/assim.batch/R/pda.mcmc.bs.R

```diff
@@ -188,9 +191,25 @@ pda.mcmc.bs <- function(settings, params.id = NULL, param.names = NULL, prior.id
         ## Read model outputs
         model.out <- pda.get.model.output(settings, run.id, bety, inputs)
         
-        ## Calculate likelihood (and store in database)
-        LL.new <- pda.calc.llik(settings, con, model.out, run.id, inputs, llik.fn)
+        # handle bias parameters if multiplicative Gaussian is listed in the likelihoods
+        if(any(unlist(any.mgauss) == "multipGauss")) {
+          isbias <- which(unlist(any.mgauss) == "multipGauss")
+          # testing now
+          nbias <- 1
+          bias.list <- return.bias(isbias, list(model.out), inputs, prior.list, nbias)
+          bias.terms <- bias.list$bias.params
+        } else {
+          bias.terms <- matrix(1, nrow = 1, ncol = 1) # just 1 for Gaussian
+        }
         
+        ## calculate error statistics      
```

Comment:
Ok, so within the brute-force MCMC that's not required, but it is an option if you want to accelerate convergence. Pseudo-code is more like:
```
chain
  iteration
     propose.parameters | bias, variances
        run.model
        get.model.output
            pda.calc.error
            pda.calc.llik.par
            pda.calc.llik
    accept/reject model parameters

    propose bias | parameters, variances
      pda.calc.*
    accept/reject bias

    gibbs sample variances | parameters, bias
```
where you are definitely free to iterate over the bias and variance sampling multiple times (or equivalently, to skip model parameter proposals), you'd just need to decide whether you store all the bias and variance samples or just the 'final' iteration (this is just a question of thinning -- you technically are allowed to store all the samples, but you may choose not to)